### ISSUE TITLE
Using `CTCLoss` with `create_supervised_trainer`

### ISSUE THREAD

**Issue by lintangsutawika** [pytorch/ignite#642]
I noticed that `create_supervised_trainer` doesn't have an option to include more arguments to the defined loss function. Or rather, the loss function is assumed to only have arguments `y_pred` and `y` so CTCLoss cannot be used here. Perhaps this could be a feature request?

**Comment by vfdev-5** [comment]
There several solutions to handle this depending on type of parameters you would like to insert
1) create your own `update_fn`
```python
def update_fn(engine, batch):
    model.train()
    optimizer.zero_grad()
    x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
    y_pred = model(x)
    
    y_pred_lengths = ...
    y_lengths = ...

    loss = loss_fn(y_pred, y, y_pred_lengths, y_lengths)
    loss.backward()
    optimizer.step()
    return output_transform(x, y, y_pred, loss)

trainer = Engine(update_fn)
```

2) recode loss function to reinterpret intput `(y_pred, y)` as `((y_pred, y_pred_lengths), (y, y_lengths))` and use default `create_supervised_trainer`

```python
class CTCLoss2(nn.CTCLoss):

    def forward(self, input, target):
        input, input_lengths = input
        target, target_lengths = target
        super(CTCLoss2, self).forward(input, target, input_lengths, target_lengths)
``` 


HTH

**Comment by vfdev-5** [comment]
I close this issue as solved, feel free to reopen if needed
